cmd/maya-apiserver/app/server/backup_endpoint_v1alpha1.go

```diff
@@ -0,0 +1,388 @@
+			laststat := findLastBackupStat(openebsClient, b)
+			// Update Backup status according to last-backup
+			updateBackupStatus(openebsClient, b, laststat)
+
+			// Get updated backup object
+			b, err = openebsClient.OpenebsV1alpha1().BackupCStors(bkp.Namespace).Get(bkp.Name, v1.GetOptions{})
+			if err != nil {
+				return nil, CodedError(400, fmt.Sprintf("Failed to fetch backup error:%v", err))
+			}
+		}
+	}
+
+	out, err := json.Marshal(b)
+	if err == nil {
+		bOps.resp.Write(out)
+		return nil, nil
+	}
+
+	return nil, CodedError(400, fmt.Sprintf("Failed to encode response data"))
+}
+
+// checkIfBKPPoolNodeDown will check if pool node is running or not
+func checkIfBKPPoolNodeDown(k8sclient *kubernetes.Clientset, bkp *v1alpha1.BackupCStor) bool {
+	var nodeDown = true
+
+	pod := findPodFromCStorID(k8sclient, bkp.Labels["cstorpool.openebs.io/uid"])
+	if pod == nil {
+		glog.Errorf("Failed to find pod for backup:%v", bkp.Name)
+		return nodeDown
+	}
+
+	if pod.Spec.NodeName == "" {
+		return nodeDown
+	}
+
+	node, err := k8sclient.CoreV1().Nodes().Get(pod.Spec.NodeName, v1.GetOptions{})
+	if err != nil {
+		glog.Infof("Failed to fetch node info for backup:%v: %v", bkp.Name, err)
+		return nodeDown
+	}
+	for _, nodestat := range node.Status.Conditions {
+		if nodestat.Type == corev1.NodeReady && nodestat.Status != corev1.ConditionTrue {
+			glog.Infof("Node:%v is not in ready state", node.Name)
+			return nodeDown
+		}
+	}
+	return !nodeDown
+}
+
+// checkIfBKPPoolPodDown will check if pool pod is running or not
+func checkIfBKPPoolPodDown(k8sclient *kubernetes.Clientset, bkp *v1alpha1.BackupCStor) bool {
+	var podDown = true
+
+	pod := findPodFromCStorID(k8sclient, bkp.Labels["cstorpool.openebs.io/uid"])
+	if pod == nil {
+		glog.Errorf("Failed to find pod for backup:%v", bkp.Name)
+		return podDown
+	}
+
+	for _, containerstatus := range pod.Status.ContainerStatuses {
+		if containerstatus.Name == "cstor-pool-mgmt" {
+			return !containerstatus.Ready
+		}
+	}
+
+	return podDown
+}
+
+// findPodFromCStorID will find the Pod having given cstorID
+func findPodFromCStorID(k8sclient *kubernetes.Clientset, cstorID string) *corev1.Pod {
+	cstorPodLabel := "app=cstor-pool"
+	podlistops := v1.ListOptions{
+		LabelSelector: cstorPodLabel,
+	}
+
+	openebsNs := os.Getenv("OPENEBS_NAMESPACE")
+	if openebsNs == "" {
+		glog.Errorf("Failed to fetch operator namespace")
+		return nil
+	}
+
+	podlist, err := k8sclient.CoreV1().Pods(openebsNs).List(podlistops)
+	if err != nil {
+		glog.Errorf("Failed to fetch pod list :%v", err)
+		return nil
+	}
+
+	for _, pod := range podlist.Items {
+		for _, env := range pod.Spec.Containers[0].Env {
+			if env.Name == "OPENEBS_IO_CSTOR_ID" && env.Value == cstorID {
+				return &pod
+			}
+		}
+	}
+	return nil
+}
+
+// findLastBackupStat will find the status of given backup from last-backup
+func findLastBackupStat(clientset internalclientset.Interface, bkp *v1alpha1.BackupCStor) v1alpha1.BackupCStorStatus {
```

Comment:
are we making sure that data is written to object storage by arc plugin by the time pool-mgmt updates BackupCstorLasts with the details?

Reply:
i will create an issue as we discussed.